odoo/addons/base/models/ir_mail_server.py

```diff
@@ -903,3 +1040,110 @@ def _onchange_encryption(self):
+        help="The organisation for which this certificate was issued.")
+    subject_alternative_names = fields.Char("Subject Alternative Names", compute='_compute',
+        help="Alternative names (usually domain names) of the organisation.")
+    issuer = fields.Char("Issuer", compute='_compute',
+        help="The certification authority (CA) that issued the certificate.")
+    not_valid_before = fields.Datetime("Not valid before", compute='_compute',
+        help="Date before which the certificate is not yet valid.")
+    not_valid_after = fields.Datetime("Not valid after", compute='_compute',
+        help="Date after which the certificate is not valid anymore.")
+    signature = fields.Char("Signature", compute='_compute',
+        help="The digital fingerprint of the certificate, signed by the certification authority.")
+
+    @api.depends('certificate_pem')
+    def _compute(self):
+        self = self.with_context(bin_size=False)
+        for record in self:
+            if not record.certificate_pem:
+                record.public_key_pem = False
+                record.subject = False
+                record.subject_alternative_names = False
+                record.issuer = False
+                record.not_valid_before = False
+                record.not_valid_after = False
+                record.signature = False
+                continue
+
+            certpem = base64.b64decode(record.certificate_pem)
+            try:
+                cert = x509.load_pem_x509_certificate(certpem)
+            except ValueError as exc:
+                record.certificate_pem = False
+                url = "https://cryptography.io/en/latest/faq.html#why-can-t-i-import-my-pem-file"
+                msg = _("Unable to load certificate. See %(url)s for more details.", url=url)
+                raise UserError(msg) from exc
+
+            record.public_key_pem = base64.b64encode(cert.public_key().public_bytes(
+                Encoding.PEM, PublicFormat.SubjectPublicKeyInfo))
+            record.subject = cert.subject.rfc4514_string()
+            record.issuer = cert.issuer.rfc4514_string()
+            record.not_valid_before = cert.not_valid_before #_utc.replace(tzinfo=None)
+            record.not_valid_after = cert.not_valid_after #_utc.replace(tzinfo=None)
+
+            # replicates openssl x509 -text behavior, for easy comparison
+            record.signature = '\n'.join(textwrap.wrap(cert.signature.hex(sep=':'), 54))
+
+            try:
+                san = cert.extensions.get_extension_for_class(x509.SubjectAlternativeName)
+            except x509.ExtensionNotFound:
+                record.subject_alternative_names = ''
+            else:
+                record.subject_alternative_names = ', '.join(
+                    san.value.get_values_for_type(x509.DNSName) +
+                    san.value.get_values_for_type(x509.IPAddress))
+
+    def save(self, mail_server=None):
+        self.ensure_one()
+        if not mail_server:
```

Comment:
Remove `mail_server` from here and add a field on the wizard.
It will be initialized from the "default_mail_server_id" context.